polars/13_window_functions.py

```diff
@@ -0,0 +1,540 @@
+    It should also only be used in a `select` context and not `with_columns`.
+
+    The result of "explode" is similar to a `group_by` followed by an `agg` followed by an `explode`.
+    """
+    )
+    return
+
+
+@app.cell
+def _(df, pl):
+    df.select(
+        pl.all().over("channel", order_by="date", mapping_strategy="explode"),
+        cumulative_revenue=pl.col("revenue")
+        .cum_sum()
+        .over("channel", order_by="date", mapping_strategy="explode"),
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""Note the modified order of the rows in the output, (but data is the same)...""")
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""## Other tips and tricks""")
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""
+    ### Reusing a window
+
+    In SQL there is a `WINDOW` keyword, which easily allows the re-use of the same window specification across expressions
+    without needing to repeat it. In Polars, this can be achieved by using `dict` unpacking to pass arguments to `over`.
+    """
+    )
+    return
+
+
+@app.cell
+def _(df_sorted, pl):
+    window = {
+        "partition_by": "date",
+        "order_by": "date",
+        "mapping_strategy": "group_to_rows",
+    }
+
+    df_sorted.with_columns(
+        pct_daily_revenue=(pl.col("revenue") / pl.col("revenue").sum()).over(**window),
+        highest_revenue_channel=pl.col("channel").top_k_by("revenue", k=1).first().over(**window),
+        daily_revenue_rank=pl.col("revenue").rank().over(**window),
+        cumulative_daily_revenue=pl.col("revenue").cum_sum().over(**window),
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""
+    ### Rolling Windows
+
+    Much like in SQL, Polars also gives you the ability to do rolling window computations. In Polars, the rolling calculation
+    is also aware of temporal data, making it easy to express if the data is not contiguous (i.e., observations are missing).
+
+    Let's look at an example of that now by filtering out one day of our data and then calculating both a 3-day and 3-row
+    max revenue split by channel...
+    """
+    )
+    return
+
+
+@app.cell
+def _(date, df, pl):
+    (
+        df.filter(pl.col("date") != date(2025, 2, 2))
+        .with_columns(
+            # "3d" -> 3 days
+            rev_3_day_max=pl.col("revenue").rolling_max_by("date", "3d", min_samples=1).over("channel"),
+            rev_3_row_max=pl.col("revenue").rolling_max(3, min_samples=1).over("channel"),
+        )
+        # sort to make the output a little easier to analyze
+        .sort("channel", "date")
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""Notice the difference in the 2nd last row...""")
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""We hope you enjoyed this notebook, demonstrating window functions in Polars!""")
+    return
+
+
+@app.cell
```

Comment:
```suggestion
@app.cell(hide_code=True)
```
Would hide the markdown cell block here.

Reply:
Agreed (and fixed)